Rules/SubscriptionValidator.cs

```diff
@@ -18,40 +18,49 @@ public class SubscriptionValidator : ISubscriptionValidator {
                 throw new ArgumentNullException(nameof(subscription));
             }
 
-            SubscriptionBase callbackParameters = null;
+
+            HttpResponseMessage response = null;
+            string challenge = Guid.NewGuid().ToString("n");
+
             if (outcome == HubValidationOutcome.Canceled) {
                 logger.LogDebug("Simulating canceled subscription.");
 
-                callbackParameters = new SubscriptionCancelled
+                SubscriptionBase callbackParameters = new SubscriptionCancelled
                 {
                     Reason = $"The subscription {subscription} was canceled for testing purposes.",
                 };
             } else {
                 logger.LogDebug("Verifying subscription.");
 
-                callbackParameters = new SubscriptionVerification
+                SubscriptionVerification callbackParameters = new SubscriptionVerification
                 {
                     // Note that this is not necessarily cryptographically random/secure.
-                    Challenge = Guid.NewGuid().ToString("n"),
-                };
-            }
+                    Challenge = challenge,
+                    LeaseSeconds = subscription.LeaseSeconds,
+                    Callback = subscription.Callback,
+                    Events = subscription.Events,
+                    Mode = subscription.Mode,
+                    Topic = subscription.Topic,
+                 };
 
-            // Default parametres for both cancel/verify.
-            callbackParameters.Callback = subscription.Callback;
-            callbackParameters.Events = subscription.Events;
-            callbackParameters.Mode = subscription.Mode;
-            callbackParameters.Topic = subscription.Topic;
+                logger.LogDebug($"Calling callback url: {subscription.Callback}");
+                string verifyUrl = subscription.Callback + "?" +                   

```

Comment:
You moved the logic for constructing the URL from the [`SubscriptionCallback`][subcallback] class and in here. Keep it in that class instead if you need to re-write it. A better commit message on why you're doing this re-write would be great.

[subcallback]: https://github.com/fhircast/sandbox/blob/master/FHIRCastSandbox/Core/SubscriptionCallback.cs